sysml.library/Kernel Library/Objects.kerml

```diff
@@ -56,4 +67,81 @@ package Objects {
 	 */
 	abstract feature binaryLinkObjects: BinaryLinkObject[0..*] nonunique subsets binaryLinks, linkObjects;
 	
+
+	/**
+	 * A Body is an Object of inner space dimension 3.
+	 */
+	struct all Body specializes Object {
+		feature redefines innerSpaceDimension = 3;
+	}
+
+	/**
+	 * A Surface is an Object of inner space dimension 2.
+	 */
+	struct all Surface specializes Object {
+		feature redefines innerSpaceDimension = 2;
+		  /* The number of  "holes" in this Surface, assuming it isClosed. */
+		feature genus : Integer[0..1] default 0;
+
+		inv { notEmpty(genus) implies (isClosed & genus >= 0) }
+	}
+
+	/*
+	 * A Curve is an Object of inner space dimension 1.
+	 */
+	struct all Curve specializes Object {
+		feature redefines innerSpaceDimension = 1;
+	}
+
+	/*
+	 * A Point is an Object of inner space dimension 0.
+	 */
+	struct all Point specializes Object {
+		feature redefines innerSpaceDimension = 0;
+	}
+
+	/*
+	 * A StructuredSpaceObject is an Object that is broken up into smaller structured space objects (cells) of
+	 * the same or lower inner space dimension: faces that are surfaces, edges that are curves, and verticies
+	 * that are points, with edges and verticies on the boundary of faces, and verticies on the boundary of
+	 * edges. Cells overlap when a structured space object is closed, as required to be a space boundary of
+	 * an object (faces overlap at their edges and/or verticies, while edges overlap at their verticies). The
+	 * inner space dimension of structured space object is the highest of their cells.
+	 */
+	abstract struct StructuredSpaceObject specializes Object {
```

Comment:
According to OED and Merriam-Webster the spelling 'vertices' was correct. So I think the last change should be reverted.

Reply:
No further comments.

Reply:
Thx (I'd spelled it two ways, for code and comments!).  See GIT question in email.